os/arch/arm/src/common/up_internal.h

```diff
@@ -223,6 +222,31 @@ extern "C" {
 
 EXTERN volatile uint32_t *current_regs;
 
+
+/* ARM requires at least a 4-byte stack alignment.  For use with EABI and
+ * floating point, the stack must be aligned to 8-byte addresses.
+ */
+
+#ifndef CONFIG_STACK_ALIGNMENT
+
+/* The symbol  __ARM_EABI__ is defined by GCC if EABI is being used.  If you
+ * are not using GCC, make sure that CONFIG_STACK_ALIGNMENT is set correctly!
+ */
+
+#ifdef __ARM_EABI__
+#define CONFIG_STACK_ALIGNMENT 8
+#else
+#define CONFIG_STACK_ALIGNMENT 4
+#endif
+#endif
```

Comment:
These changes look good. However, to make it more accurate and complete, we could change the declaration of interrupt stacks in the exception.S file. Currently, the interrupt stack is always aligned to 4 bytes as follows. But it must be either 4 or 8 bytes. 

```
        .align  4
g_nestedirqstkalloc:
        .skip   (CONFIG_ARCH_NESTED_IRQ_STACK_SIZE & ~3)
g_nestedirqstkbase:

```